statsmodels/stats/regularized_covariance.py

```diff
@@ -0,0 +1,179 @@
+    -------
+    A scalar
+
+    Notes
+    -----
+
+    nodewise_weight_i = sqrt(1/n ||wexog,i - wexog_-i nodewise_row||_2^2
+                             + alpha ||nodewise_row||_1)
+    """
+
+    n, p = wexog.shape
+    # handle array alphas
+    if not np.isscalar(alpha):
+        alpha = alpha[ind]
+
+    ind = list(range(p))
+    ind.pop(idx)
+
+    d = np.linalg.norm(wexog[:, idx] - wexog[:, ind].dot(nodewise_row))**2
+    d = np.sqrt(d / n + alpha * np.linalg.norm(nodewise_row, 1))
+    return d
+
+
+def _calc_approx_inv_cov(nodewise_row_l, nodewise_weight_l):
+    """calculates the approximate inverse covariance matrix
+
+    Parameters
+    ----------
+    nodewise_row_l : list
+        A list of array-like object where each object corresponds to
+        the nodewise_row values for the corresponding variable, should
+        be length p.
+    nodewise_weight_l : list
+        A list of scalars where each scalar corresponds to the nodewise_weight
+        value for the corresponding variable, should be length p.
+
+    Returns
+    ------
+    An array-like object, p x p matrix
+
+    Notes
+    -----
+
+    nwr = nodewise_row
+    nww = nodewise_weight
+
+    approx_inv_cov_j = - 1 / nww_j [nwr_j,1,...,1,...nwr_j,p]
+    """
+
+    p = len(nodewise_weight_l)
+
+    approx_inv_cov = np.eye(p)
+    for idx in range(p):
+        ind = list(range(p))
+        ind.pop(idx)
+        approx_inv_cov[idx,ind] = nodewise_row_l[idx]
+    approx_inv_cov *= -1 / nodewise_weight_l[:, None]**2
+
+    return approx_inv_cov
+
+
+class RegularizedInvCovariance(object):
+    __doc__ = """
+    Class for estimating regularized inverse covariance with
+    nodewise regression
+
+    Parameters
+    ----------
+    wexog : array-like
+        A weighted design matrix for covariance
+    alpha : scalar
+        Regularizing constant
+
+    Attributes
+    ----------
+    wexog : array-like
+        A weighted design matrix for covariance
+    alpha : scalar
+        Regularizing constant
+    """
+
+    def __init__(self, wexog, alpha):
+
+        self.wexog = wexog
+        self.alpha = alpha
+
+
+    def fit_nodewise(self):
+        """estimates the regularized inverse covariance using nodewise
+        regression
+
+        Parameters
+        ----------
+        """
+
+        n, p = self.wexog.shape
+
+        nodewise_row_l = []
+        nodewise_weight_l = []
+
+        for idx in range(p):
+            nodewise_row = _calc_nodewise_row(self.wexog, idx, self.alpha)
+            nodewise_row_l.append(nodewise_row)
+
+            nodewise_weight = _calc_nodewise_weight(self.wexog, nodewise_row,
+                                                    idx, self.alpha)
+            nodewise_weight_l.append(nodewise_weight)
+
+        nodewise_row_l = np.array(nodewise_row_l)
+        nodewise_weight_l = np.array(nodewise_weight_l)
+
+        approx_inv_cov = _calc_approx_inv_cov(nodewise_row_l, nodewise_weight_l)
+
+        return approx_inv_cov
```

Comment:
I think the more consistent way to do this is to not return anything, just attach the result to the class (call it something shorter and private, like `_cov_inv`).

Then have an accessor function

```
@cache_readonly
def cov_inv(self):
    return self._cov_inv
```